pkg/store/proxy.go

```diff
@@ -344,6 +355,45 @@ func startStreamSeriesSet(
 
 	wg.Add(1)
 	go func() {
+		const (
+			// approximate the length of each label being about 20 chars, e.g. "k8s_app_metric0"
+			approxLabelLen = int64(10)
+
+			// approximate the size if chunks by having 120 bytes in each?
+			approxChunkLen = int64(120)
+		)
+
+		// buffer query sizes and process in chunks of 20MB or so
+		// to avoid hammering cache lines with atomic increments.
+		const querySizeBufferSize = 20 * 1000 * 1000
+		querySizeBuffer := int64(0)
+		queryLocalSize := int64(0)
+		queryTotalSize := *queryTotalSizeRef
+
+		defer func() {
+			totalSizeMsg := fmt.Sprintf("%.2fMB", float64(queryTotalSize)/float64(1000000))
+			localSizeMsg := fmt.Sprintf("%.2fMB", float64(queryLocalSize)/float64(1000000))
```

Comment:
We could probably reuse `byteCountToHuman` here.

Reply:
Yes, but I don't want to expose `byteCountToHuman` as it's really an internal package detail, it doesn't make sense for the package to provide `ByteCountToHuman` I think.

If this bothers anyone, maybe indeed taking dep on humanize makes sense and then we can use it in both places. I don't know if benefits of extra dep outweight the cost of this though.

Not too fussed about either approach if this is what blocks this PR :)